src/pages/LenderProfile/LenderProfile.vue

```diff
@@ -151,6 +192,14 @@ export default {
 		this.lenderInfo = cachedLenderInfo?.community?.lender ?? {};
 		this.allAchievements = cachedLenderInfo?.userAchievementProgress?.achievementProgress ?? [];
 		this.lenderIsPublic = !!this.lenderInfo?.id;
+
+		const utmCampaign = this.$route?.query?.utm_campaign ?? '';
+		const isUtmValid = utmCampaign.includes('badge_') && utmCampaign.includes('social_share');
+		const badgeKey = utmCampaign.split('badge_')[1];
+		const isBadgeKeyValid = defaultBadges.includes(badgeKey);
```

Comment:
There's some duplicated logic here to check the badge, can it be extracted to a method and be used in both places? Similarly, will this logic be needed in the home page as well? If so, most of the new code here can be extracted to a new composable (yay Vue3!) and used here and in the home page.